gammapy/datasets/simulate.py

```diff
@@ -26,30 +28,164 @@ class MapDatasetEventSampler:
             energy = coords["energy"]
 
+        table["TIME"] = u.Quantity(((time.mjd - time_ref.mjd) * u.day).to(u.s)).to("s")
         table["ENERGY_TRUE"] = energy
         table["RA_TRUE"] = coords.skycoord.icrs.ra.to("deg")
         table["DEC_TRUE"] = coords.skycoord.icrs.dec.to("deg")
 
+        return table
+
+    def __evaluate_timevar_source(self, dataset, evaluator, time_axis=None):
+        """Calculate Npred for a given `dataset.model` by evaluating
+        it in region geometry.
+
+        Parameters
+        ----------
+        dataset : `~gammapy.datasets.MapDataset`
+            Map dataset.
+        time_axis : `~gammapy.Maps.MapAxis`
+            Axis of the time.
+
+        Returns
+        -------
+        dataset.npred() : `~gammapy.dataset.MapDataset`
+            Npred map.
+        """
+        energy_axis = MapAxis.from_edges(
+            dataset.geoms["geom"].axes["energy"].edges, name="energy_true"
+        )
+
+        target = evaluator.model.spatial_model.position
+        on_region = PointSkyRegion(center=target)
+        region_geom = RegionGeom.create(on_region, axes=[energy_axis])
+
+        flux = evaluator.model.evaluate_geom(region_geom.to_wcs_geom(), gti=dataset.gti)
+        region_exposure = dataset.exposure.to_region_nd_map(region_geom.center_skydir)
+
+        npred = flux * region_exposure * dataset.geoms["geom"].bin_volume()
+        dataset.npred().data = npred.value
+
+        return dataset.npred()
+
+    def _sample_coord_time_energy(self, dataset, evaluator, t_delta="1 s"):
+        """Sample model components of a time-varying source.
+
+        Parameters
+        ----------
+        dataset : `~gammapy.datasets.MapDataset`
+            Map dataset.
+        t_delta : ~str
+            Minimum step time.
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table of sampled events.
+        """
+        if evaluator.model.spatial_model.tag[0] != "PointSpatialModel":
+            raise ValueError(
+                "Event sampler cannot work for time-varying extended sources."
+            )
+
+        time_start, time_stop, time_ref = (
+            evaluator.gti.time_start,
+            evaluator.gti.time_stop,
+            evaluator.gti.time_ref,
+        )
+        t_min = Time(time_start)
+        t_max = Time(time_stop)
+        t_delta = u.Quantity(t_delta)
+
+        ontime = u.Quantity((t_max - t_min).sec, "s")
+
+        time_unit = (
+            u.Unit(evaluator.model.temporal_model.table.meta["TIMEUNIT"])
+            if hasattr(evaluator.model.temporal_model, "table")
+            else ontime.unit
+        )
+
+        t_step = t_delta.to_value(time_unit)
+        t_step = (t_step * u.s).to("d")
+
+        #        t = Time(np.arange(t_min.mjd, t_max.mjd, t_step.value), format="mjd")
+        #        time_axis = MapAxis.from_edges(t.value * u.d, name="Time")
+
+        npred = self.__evaluate_timevar_source(dataset, evaluator)
```

Comment:
Why the `__` ? The function does not exist yet?

Reply:
```suggestion
        npred = self._evaluate_timevar_source(dataset, evaluator)
```